php_phongo.c

```diff
@@ -1591,6 +1607,178 @@ void php_phongo_populate_default_ssl_ctx(php_stream_context *ctx, zval *driverOp
 #undef SET_STRING_CTX
 } /* }}} */
 
+bool php_phongo_apply_rp_options_to_client(mongoc_client_t *client, bson_t *options TSRMLS_DC) /* {{{ */
+{
+	bson_iter_t iter;
+	mongoc_read_prefs_t *new_rp;
+	const mongoc_read_prefs_t *old_rp;
+
+	if (!(old_rp = mongoc_client_get_read_prefs(client))) {
+		MONGOC_WARNING("Client does not have a read preference.");
```

Comment:
This is no difference than the warnings raised by libmongoc during URI parsing, though. They end up logged (if logging is enabled) or ignored. I can certainly remove these if you think it has no value, though.

This particular line (checking `old_rp`) is more of an assertion, because the client should always have a valid RP and WC struct.